src/ia-account-settings.ts

```diff
@@ -0,0 +1,804 @@
+    this.userData.password = input.value;
+    this.fieldsError.password = '';
+    this.resetErrorFields('password');
+  }
+
+  /** @inheritdoc */
+  setBorrowHistory(e: Event) {
+    const input = e.target as HTMLInputElement;
+    this.loanHistoryFlag = input.checked ? 'public' : 'private';
+    this.changeSaveButtonState();
+  }
+
+  /** @inheritdoc */
+  setMailingList(e: Event) {
+    const input = e.target as HTMLInputElement;
+    const fieldName = input.name;
+
+    if (input.checked) {
+      this.selectedMailingLists.push(fieldName);
+    } else {
+      const index = this.selectedMailingLists.indexOf(fieldName);
+      this.selectedMailingLists.splice(index, 1);
+    }
+
+    this.changeSaveButtonState();
+  }
+
+  setLinkedProvider(e: Event) {
+    const input = e.target as HTMLInputElement;
+    const { provider } = input.dataset;
+    if (!provider) {
+      return;
+    }
+
+    if (input.checked) {
+      const index = this.unlinkProviders?.indexOf(provider);
+      this.unlinkProviders?.splice(index, 1);
+    } else {
+      this.unlinkProviders?.push(provider);
+    }
+
+    this.changeSaveButtonState();
+  }
+
+  changeSaveButtonState() {
+    this.saveButtonDisabled = !!this.hasFieldError();
+  }
+
+  /** @inheritdoc */
+  resetErrorFields(field: string) {
+    let errorFields = {};
+
+    if (field === 'email') {
+      errorFields = { ...this.fieldsError, email: '' };
+    } else if (field === 'screenname') {
+      errorFields = { ...this.fieldsError, screenname: '' };
+    } else {
+      errorFields = { ...this.fieldsError, password: '' };
+    }
+
+    this.saveButtonDisabled = false;
+    this.fieldsError = errorFields;
+  }
+
+  /** @inheritdoc */
+  hasFieldError() {
+    return Object.values(this.fieldsError).some(value => value !== '');
+  }
+
+  /** @inheritdoc */
+  async validateScreenname() {
+    let error = '';
+    this.userData.screenname = trimString(this.userData.screenname as string);
+
+    if (this.userData.screenname === '') {
+      error = 'Screen name can not be empty.';
+    } else if (this.userData.screenname?.match(/[\\]/gm)) {
+      error = 'Invalid screen name';
+    } else if ((await this.isScreennameAvailable()) === false) {
+      error = 'This screen name is already being used by another user.';
+    }
+
+    this.fieldsError = { ...this.fieldsError, screenname: error };
+  }
+
+  /** @inheritdoc */
+  async validateEmail() {
+    let error = '';
+    this.userData.email = trimString(this.userData.email as string);
+
+    const emailRegex =
+      /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/;
+
+    if (this.userData.email === '') {
+      error = 'Email address can not be empty.';
+    } else if (!this.userData.email?.match(emailRegex)) {
+      error = 'Email address contains invalid characters and/or whitespace.';
+    } else if (await !this.isEmailAvailable()) {
+      error = 'This email address is already being used by another user.';
```

Comment:
Blocker: the `await` needs to be applied directly to the `this.isEmailAvailable()` result (like with the screen name above). Right now this just negates the Promise object itself, so this condition will always be false.